cumulusci/tasks/bulkdata/tests/test_step.py

```diff
@@ -879,6 +1109,452 @@ def test_get_prev_record_values(self):
+                )
+            )
+            == 1
+        )
+
+    @responses.activate
+    def test_select_records_random_similarity_success__pagination(self):
+        mock_describe_calls()
+        task = _make_task(
+            LoadData,
+            {
+                "options": {
+                    "database_url": "sqlite:///test.db",
+                    "mapping": "mapping.yml",
+                }
+            },
+        )
+        task.project_config.project__package__api_version = CURRENT_SF_API_VERSION
+        task._init_task()
+
+        responses.add(
+            responses.POST,
+            url=f"https://example.com/services/data/v{CURRENT_SF_API_VERSION}/composite/sobjects",
+            json=[
+                {"id": "003000000000001", "success": True},
+                {"id": "003000000000002", "success": True},
+            ],
+            status=200,
+        )
+        responses.add(
+            responses.POST,
+            url=f"https://example.com/services/data/v{CURRENT_SF_API_VERSION}/composite/sobjects",
+            json=[{"id": "003000000000003", "success": True}],
+            status=200,
+        )
+        step = RestApiDmlOperation(
+            sobject="Contact",
+            operation=DataOperationType.UPSERT,
+            api_options={"batch_size": 10, "update_key": "LastName"},
+            context=task,
+            fields=["Id", "Name", "Email"],
+            selection_strategy=SelectStrategy.SIMILARITY,
+        )
+
+        results = {
+            "records": [
+                {
+                    "Id": "003000000000001",
+                    "Name": "Jawad",
+                    "Email": "mjawadtp@example.com",
+                },
+            ],
+            "done": False,
+            "nextRecordsUrl": "https://example.com",
+        }
+        results_more = {
+            "records": [
+                {
+                    "Id": "003000000000002",
+                    "Name": "Aditya",
+                    "Email": "aditya@example.com",
+                },
+                {
+                    "Id": "003000000000003",
+                    "Name": "Tom Cruise",
+                    "Email": "tomcruise@example.com",
+                },
+            ],
+            "done": True,
+        }
+        step.sf.query = mock.Mock()
+        step.sf.query.return_value = results
+        step.sf.query_more = mock.Mock()
+        step.sf.query_more.return_value = results_more
+        records = iter(
+            [
+                ["Id: 1", "Jawad", "mjawadtp@example.com"],
+                ["Id: 2", "Aditya", "aditya@example.com"],
+                ["Id: 2", "Tom", "tom@example.com"],
+            ]
+        )
+        step.start()
+        step.select_records(records)
+        step.end()
+
+        # Get the results and assert their properties
+        results = list(step.get_results())
+        assert len(results) == 3  # Expect 3 results (matching the input records count)
```

Comment:
Add the same asserts as before for each id instead of asserting the length of records as they will always be 3

Reply:
Done